Source/source/GPS.cpp

```diff
@@ -60,6 +62,41 @@ string toStringTime(struct minmea_time *time) {
 	return timeString.str();
 }
 
+
+unsigned int encode_time_as_int(struct minmea_time *time) {
+    unsigned int BITS = 32;
+    unsigned int H_BITS = 5;
+    unsigned int M_BITS = 6;
+    unsigned int S_BITS = 6;
+    unsigned int enc = 0;
+    unsigned int hours = (unsigned int)(time->hours);
+    unsigned int minutes = (unsigned int)(time->minutes);
+    unsigned int seconds = (unsigned int)(time->seconds);
+    
+    enc = enc | hours << BITS - H_BITS >> BITS - H_BITS - M_BITS - S_BITS;
+    enc = enc | minutes << BITS - M_BITS >> BITS - M_BITS - S_BITS;
+    enc = enc | seconds << BITS - S_BITS >> BITS - S_BITS;
+    
+    return enc;
+}
+
+
+struct minmea_time decode_time_from_int(unsigned int time_int) {
+    unsigned int hours, minutes, seconds;
+    unsigned int BITS = 32;
+    unsigned int H_BITS = 5;
+    unsigned int M_BITS = 6;
+    unsigned int S_BITS = 6;
+    struct minmea_time time = {};
+    
+    time.seconds = time_int << BITS - S_BITS >> BITS - S_BITS;
+    time.minutes = time_int << BITS - M_BITS - S_BITS >> BITS - M_BITS;
+    time.hours = time_int << BITS - H_BITS - M_BITS - S_BITS >> BITS - H_BITS;
```

Comment:
Ditto as above, parentheses would add to code clarity

Reply:
Added for both in next commit.